### PR TITLE
Adding Header Bar to the application

### PR DESCRIPTION
**Pull request by vdorceus-excilys**
Added Header Support for the entire application

### CODE REVIEW

File: src/app/header/header.component.html [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<mat-toolbar color='primary'>
+  <mat-toolbar-row>
+    <div class='container'>
+        <div class='row'>
+          <div class='col-md-3'>
+            <span>Computer Management System</span>
+          </div>
+          <div class='col-md-3'></div>
```
Comment (RoudautTanguy) [inline src/app/header/header.component.html]: Use 'offset-md-6' instead of two empty div

File: src/app/header/header.component.html [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+          <div class='col-md-3'></div>
+          <div class='col-md-3'>
+              <span>
+                  <button mat-icon-button [matMenuTriggerFor]="menu" aria-label="Example icon-button with a menu">
```
Comment (RoudautTanguy) [inline src/app/header/header.component.html]: Should not be here

File: src/app/list-company/list-company.component.html [hunk 3]
```diff
@@ -1,3 +1,27 @@
-<p>
-  list-company works!
-</p>
+<mat-form-field>
```
Comment (RoudautTanguy) [inline src/app/list-company/list-company.component.html]: Add a container div so the filter input and the list don't take all the place available

**Comment by vdorceus-excilys** [comment]
Try it out !  I fixed most of the issues

File: src/app/list-company/list-company.component.ts [hunk 4]
```diff
@@ -1,15 +1,60 @@
...
 })
+
+/**
+ * @title Data table with sorting, pagination, and filtering.
+ */
+
 export class ListCompanyComponent implements OnInit {
 
-  constructor() { }
+  displayedColumns: string[] = ["id","name"];
```
Comment (RoudautTanguy) [inline src/app/list-company/list-company.component.ts]: " should be ' (quotemark)

File: src/app/list-company/list-company.component.ts [hunk 5]
```diff
@@ -1,15 +1,60 @@
...
+  dataSource: MatTableDataSource<CompanyDTO>;
+  @ViewChild(MatPaginator,{static: true}) paginator: MatPaginator;
+  @ViewChild(MatSort, {static: true}) sort: MatSort;
+
+  constructor() { 
+    // Create 100 users
+    const companyList = Array.from({length: 100}, (_, k) => createCompany(k + 1));
```
Comment (RoudautTanguy) [inline src/app/list-company/list-company.component.ts]: Use the API

File: src/app/list-computer/list-computer.component.html [hunk 6]
```diff
@@ -1,3 +1,44 @@
...
+      <td mat-cell *matCellDef="let row"> {{row.introduced}} </td>
+    </ng-container>
+
+    <!-- Discontinued Column -->
+    <ng-container matColumnDef="discontinued">
+      <th mat-header-cell *matHeaderCellDef mat-sort-header> Discontinued </th>
+      <td mat-cell *matCellDef="let row" > {{row.discontinued}} </td>
+    </ng-container>
+
+      <!-- Discontinued Column -->
```
Comment (RoudautTanguy) [inline src/app/list-computer/list-computer.component.html]: Wrong comment

File: src/app/list-computer/list-computer.component.ts [hunk 7]
```diff
@@ -1,15 +1,66 @@
...
 
-  constructor() { }
+  displayedColumns: string[] = ["id","name","introduced","discontinued","company"];
+  dataSource: MatTableDataSource<ComputerDTO>;
+  @ViewChild(MatPaginator,{static: true}) paginator: MatPaginator;
+  @ViewChild(MatSort, {static: true}) sort: MatSort;
+
+  constructor() { 
+    // Create 100 users
+    const computerList = Array.from({length: 100}, (_, k) => createNewUser(k + 1));
```
Comment (RoudautTanguy) [inline src/app/list-computer/list-computer.component.ts]: Use the API

File: src/app/header/header.component.html [hunk 8]
```diff
@@ -5,8 +5,7 @@
           <div class='col-md-3'>
             <span>Computer Management System</span>
           </div>
-          <div class='col-md-3'></div>
-          <div class='col-md-3'></div>
+          <div class='col-md-6'></div>
```
Comment (RoudautTanguy) [inline src/app/header/header.component.html]: Not yet ;) https://getbootstrap.com/docs/4.0/layout/grid/#offset-classes
Add offset-md-6 to the div with content

**Pull request merged**